test/fika/type_checker_test.exs

```diff
@@ -508,6 +507,50 @@ defmodule Fika.TypeCheckerTest do
       assert {:ok, :Bool, _} = TypeChecker.infer(function, env)
     end
 
+    test "when function accepts union types and has if-else clause" do
```

Comment:
What do we mean when we say "function accepts union types"? If we are talking about the expected return type, we can say "function is expected to return union types"